homeassistant/components/cmus/media_player.py

```diff
@@ -65,33 +65,66 @@ def setup_platform(hass, config, add_entities, discover_info=None):
     name = config[CONF_NAME]
 
     try:
-        cmus_remote = CmusDevice(host, password, port, name)
+        cmus_remote = CmusRemote(server=host, port=port, password=password)
+        cmus_remote.connect()
+
+        if cmus_remote.cmus is None:
+            return
+
     except exceptions.InvalidPassword:
```

Comment:
This will never happen now since we catch the error inside the `CmusRemote` class.

Reply:
Good point! I just removed the redundant try-catch here. Much cleaner...